src/main/java/frc/robot/commands/ReleaseAlgae.java

```diff
@@ -0,0 +1,29 @@
+package frc.robot.commands;
+
+import com.spikes2212.command.drivetrains.commands.DriveArcadeWithPID;
+import com.spikes2212.command.genericsubsystem.commands.MoveGenericSubsystem;
+import edu.wpi.first.wpilibj.Timer;
+import frc.robot.subsystems.Gripper;
+
+public class ReleaseAlgae extends MoveGenericSubsystem {
+
+    private static final double TIME_TO_RELEASE = 0.5;
+    private static final double RELEASE_SPEED = 0.5;
+
+    private final Gripper gripper;
+
+    private double lastTimeInGripper = 0;
+
+    public ReleaseAlgae(Gripper gripper) {
+        super(gripper, RELEASE_SPEED);
+        this.gripper = gripper;
+    }
+
+    @Override
+    public boolean isFinished() {
+        if (gripper.hasAlgae()) {
+            lastTimeInGripper = Timer.getFPGATimestamp();
+        }
+        return Timer.getFPGATimestamp() - lastTimeInGripper >= TIME_TO_RELEASE;
```

Comment:
this function is called each command iteration, so once every couple of milliseconds.
each time it would check if the gripper has an Algea, set the last time to `now` if it does, and check if `now - now` is greater than `TIME_TO_RELEASE`. which is not, because `now - now = 0`

you can make the `lastTimeInGripper` an attribute in the class, so the last line of code would work.

another way is to initiate `startTime = Timer.getFPGATimestamp()` in the `initialize()` of the command, and check if the delta between the start time and current time is the `TIME_TO_RELEASE`